### ISSUE TITLE
Export Hockley in local dev environment due to 504 error

### ISSUE THREAD

**Issue by lalgee** [LibraryOfCongress/concordia#1601]
**What behavior did you observe? Please describe the bug**
"Seers, Spiritualists, and the Spirit World: The experiments of Frederick Hockley" export times out in production as well as test. A 504 error eventually loads. The bag does not download and a new bag is not created in AWS.

This bug is documented in https://github.com/LibraryOfCongress/concordia/issues/1503.  Established workaround is to export in a local development environment.

**How can we reproduce the bug?**

1. Go to Django Admin
2. Go to Campaigns
3. Click on Seers, Spiritualists, and the Spirit World: The experiments of Frederick Hockley
4. Click EXPORT BAGIT
5. See error
6. What is the expected behavior?
7. After clicking the EXPORT BAGIT button a .zip file containing all campaign transcription TXTs and metadata should download.

**Comment by jkueloc** [comment]
Several attempts to run the export BagIt for Hockley - the multi-volume import - from my local mac dev environment resulted in my local mac's swap size to rise > 50GB over time and my available storage drop from 87 GB to 28 GB and back up to ~31 GB before the OS would kill the process.   There is one item and 4652 assets in the campaign.

Larger campaigns like Mary Church Terrell export BagIt successfully even if it took a several minutes.

I tried changing a Mary Church Terrell asset download url to a Hockley url. However, Mary Church Terrell campaign successfully export BagIt with the one Hockley folder structure and transcript.
 
When the export BagIt process runs it uses the **item.metadata value** to look for and write out an item_resource_url for the item to the item-resource-urls.txt file in the BagIt data directory.  There are a lot of duplicated records in this file. My understanding is that only the unique values are used in downstream out of Concordia processing.

For Hockley, there is ~ 7.3**MB** of data in the item.metadata field when written to file. Mary Church Terrell there is ~ 270 **KB** for 1 of the 7 items. 

To speed up the process of elimination I commented out the code the call to write_item_resource_file that generates and writes the item-resource-urls.txt file and was able to successfully run the Campaign export BagIt for Mary Church Terrell ad Hockley.

The BagIt zip was reviewed by @laurenseroka and that's all that was needed and the distinct 11 resource_urls list can be manually generated.

It would seem this chunk of code's purpose needs to be better understood. The Asset records have the same resource_url value albeit with 8 extra trailing characters.

**Comment by jkueloc** [comment]
Question for @laurenseroka and @lalgee , @elainekamlley  - I'm looking for confirmation. 

Is the sole purpose of the item-resource-urls.txt file in Concordia to be used downstream by a human in the loop as a resource to create a list of unique resource urls for the campaign, project or item export BagIt asset contents?

**Comment by laurenseroka** [comment]
You are correct in that we use these text files to get the unique resource urls for a given set of content. We use them as a way to create the concatenated files once the content has been exported. 
Maybe @lalgee can think other ways these are used, or maybe the original intention of the structure of this file.

**Comment by lalgee** [comment]
I don't know of any of any other use for this data.  I wasn't involved directly in its design, but export was structured based on the dev team's understanding of CTS/ETL and the existing workflows at the time. I don't believe there was any intended use outside of serving the loc.gov workflow, but @elainekamlley may know more.
